### PR TITLE
CDAP-15838 Decompress-action plugin validation

### PR DESCRIPTION
**Pull request by mlozbin-cybervisiontech**
JIRA: https://issues.cask.co/browse/CDAP-15838
WIKI: https://wiki.cask.co/display/CE/Plugin+Validation

In scope of this PR:
- integrated with new validation API

### CODE REVIEW

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau @CuriousVini could you, please, take a look.

File: src/main/java/io/cdap/plugin/decompress/action/DecompressActionConfig.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+  /**
+   * Validates the config parameters required for unloading the data.
+   */
+  public void validate(FailureCollector collector) {
+    try {
+      Pattern.compile(getFileRegex());
+    } catch (Exception e) {
+      collector.addFailure("The regular expression pattern provided is not a valid " +
+                                           "regular expression.", null)
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/decompress/action/DecompressActionConfig.java]: nit: fix indentation

**Review by albertshau — APPROVED** [review]
one nitpick on indentation. please fix and squash commits.

File: src/main/java/io/cdap/plugin/decompress/action/DecompressActionConfig.java (hunk 1 above)
Comment (mlozbin-cybervisiontech) [reply src/main/java/io/cdap/plugin/decompress/action/DecompressActionConfig.java]: done

**Comment by mlozbin-cybervisiontech** [comment]
@albertshau I addressed all your comments and squashed commits, could you, please, merge.

**Pull request merged**